File: core/src/main/java/com/ustadmobile/core/view/SelectPeopleDialogView.java
package com.ustadmobile.core.view;

import com.ustadmobile.core.db.UmProvider;
import com.ustadmobile.lib.db.entities.PersonWithEnrollment;

public interface SelectPeopleDialogView extends UstadView {

    String VIEW_NAME="SelectPeopleDialogView";
    String ARG_SELECTED_PEOPLE = "SelectedPeople";
    String ARG_SELECT_ACTOR = "SelectActor";
    void finish();

    void setPeopleProvider(UmProvider<PersonWithEnrollment> peopleListProvider);
}

File: core/src/commonMain/kotlin/com/ustadmobile/core/view/SelectPeopleDialogView.kt
package com.ustadmobile.core.view

import androidx.paging.DataSource
import com.ustadmobile.core.db.UmProvider
import com.ustadmobile.lib.db.entities.PersonWithEnrollment

interface SelectPeopleDialogView : UstadView {
    fun finish()

    fun setPeopleProvider(peopleListProvider: DataSource.Factory<Int, PersonWithEnrollment>)

    companion object {

        val VIEW_NAME = "SelectPeopleDialogView"
        val ARG_SELECTED_PEOPLE = "SelectedPeople"
        val ARG_SELECT_ACTOR = "SelectActor"
    }
}
